pkg/controllers/provisioning/v1alpha1/reallocation/controller.go

```diff
@@ -87,24 +90,39 @@ func (c *Controller) Reconcile(object controllers.Object) error {
 	}
 
 	// 4. Find Nodes Past TTL with Karpenter Labels
-	expired, err := c.filter.GetExpiredNodes(ctx, provisioner)
+	terminable, err := c.filter.GetTerminableNodes(ctx, provisioner)
 	if err != nil {
-		return fmt.Errorf("getting expired nodes, %w", err)
+		return fmt.Errorf("getting terminable nodes, %w", err)
 	}
-	if len(expired) == 0 {
+	if len(terminable) == 0 {
 		return nil
 	}
 
-	// 5. Cordon each node
-	if err := c.annotator.CordonNodes(ctx, c.filter.GetCordonableNodes(expired)); err != nil {
+	// 5. Cordon each node and label draining
+	if err := c.annotator.CordonNodes(ctx, c.filter.GetCordonableNodes(terminable)); err != nil {
 		return fmt.Errorf("cordoning node, %w", err)
 	}
 
-	// TODO
-	// 6. Drain Nodes past TTL
+	// 6. Drain Nodes with draining label
+	draining, err := c.filter.GetLabeledNodes(ctx, provisioner, map[string]string{
+		v1alpha1.ProvisionerDrainingKey: "draining",
+	})
+	if err != nil {
+		return fmt.Errorf("listing nodes, %w", err)
+	}
+
+	if err := c.terminator.DrainNodes(ctx, draining, provisioner); err != nil {
+		return fmt.Errorf("draining nodes, %w", err)
+	}
```

Comment:
Not clear why are we trying to drain nodes which are already labelled `draining`

Reply:
As we are calling this reconcile loop every 5 seconds, we need to constantly be evicting the pods that are on the node, or draining, since this can take any amount of time, and can fail for many reasons.